### PR TITLE
Adapting docker build sequence

### PR DESCRIPTION
**Pull request by sd4v1d**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -35,17 +35,14 @@ RUN npm run build
 #install server dep
 WORKDIR ${BASEDIR}/${EDUMEET}/server
 
-RUN apt-get install -y git build-essential python openssl libssl-dev pkg-config 
-
-RUN npm install
-RUN npm install logstash-client
+RUN npm install;npm run build;npm install logstash-client
```
Comment (misi) [inline Dockerfile:38]: I think maybe it could be spread this one lint to separated tasks. But it is not strong opinion.

File: Dockerfile [hunk 2]
```diff
@@ -3,14 +3,14 @@ FROM node:lts-buster-slim AS edumeet-builder
 # Args
 ARG BASEDIR=/opt
 ARG EDUMEET=edumeet
-ARG NODE_ENV=production
+#ARG NODE_ENV=production
```
Comment (misi) [inline Dockerfile:6]: We can leave it as it was before

File: Dockerfile [hunk 3]
```diff
@@ -21,7 +21,7 @@ WORKDIR ${BASEDIR}/${EDUMEET}/app
 RUN npm install
 
 # set app in producion mode/minified/.
-ENV NODE_ENV ${NODE_ENV}
+#ENV NODE_ENV ${NODE_ENV}
```
Comment (misi) [inline Dockerfile:24]: We can leave it as before

File: Dockerfile (hunk 1 above)
Comment (misi) [reply Dockerfile:38]: And replace `npm install` with `npm install --production=false` to install always develop dependencies this overrides the NODE_ENV setting

File: Dockerfile [hunk 4]
```diff
@@ -35,17 +35,14 @@ RUN npm run build
...
-RUN npm install logstash-client
+RUN npm install;npm run build;npm install logstash-client
 
 FROM node:lts-buster-slim
 
 # Args:
 ARG BASEDIR=/opt
 ARG EDUMEET=edumeet
-ARG NODE_ENV=production
+#ARG NODE_ENV=production
```
Comment (misi) [inline Dockerfile:45]: Revert it also

File: Dockerfile [hunk 5]
```diff
@@ -65,3 +62,4 @@ ENV DEBUG ${SERVER_DEBUG}
 
 COPY docker-entrypoint.sh /
 ENTRYPOINT ["/docker-entrypoint.sh"]
+#ENTRYPOINT ["bash"]
```
Comment (misi) [inline Dockerfile:65]: Remove this line. I guess it was for debug

File: configs/server/config.js [hunk 6]
```diff
@@ -1,4 +1,7 @@
+//serial: 2021051303
```
Comment (misi) [inline configs/server/config.js:1]: Remove this line. I guess it was for debug

File: configs/server/config.js [hunk 7]
```diff
@@ -153,9 +159,9 @@ module.exports =
 	// if you use encrypted private key the set the passphrase
 	tls          :
 	{
-		cert : `${__dirname}/../certs/cert.pem`,
+		cert : `${__dirname}/../../certs/cert.pem`,
```
Comment (misi) [inline configs/server/config.js:162]: Cert and privkey is not changed so I think you should revert this change.

File: configs/server/config.js [hunk 8]
```diff
@@ -153,9 +159,9 @@ module.exports =
...
 		// passphrase: 'key_password'
-		key  : `${__dirname}/../certs/privkey.pem`
+		key  : `${__dirname}/../../certs/privkey.pem`
```
Comment (misi) [inline configs/server/config.js:164]: Cert and privkey location is not changed so I think you should revert this change.

File: docker-entrypoint.sh [hunk 9]
```diff
@@ -1,9 +1,16 @@
-#!/bin/sh 
+#!/bin/bash
```
Comment (misi) [inline docker-entrypoint.sh:1]: Do we really need bash here? What feature was missing from sh?

File: docker-entrypoint.sh [hunk 10]
```diff
@@ -1,9 +1,16 @@
...
 set -e 
 
 cd ${BASEDIR}/${EDUMEET}/server
-node ${BASEDIR}/${EDUMEET}/server/server.js
+if [[ ! -e ${BASEDIR}/${EDUMEET}/server/dist/config/config.js ]] || [[ $(head -1 ${BASEDIR}/${EDUMEET}/server/config/config.js) != $(head -1 ${BASEDIR}/${EDUMEET}/server/dist/config/config.js) ]] 
+  then
+      echo "Rebuilding configuration..."
+      npm run build
+fi
+
```
Comment (misi) [inline docker-entrypoint.sh:12]: The new config mgmt is not yet merged to develop so let's hold back this changes until it will be merged.

**Review by misi — COMMENTED** [review]
We need some adjustments. Can you please check my comments? @sd4v1d 
THX!

File: Dockerfile (hunk 1 above)
Comment (sd4v1d) [reply Dockerfile]: Separated in three tasks. Only difference is that docker creates separate layer for each command.  
`npm install` replaced with `npm install --production=false`

File: Dockerfile (hunk 2 above)
Comment (sd4v1d) [reply Dockerfile]: Reverted.

File: Dockerfile (hunk 3 above)
Comment (sd4v1d) [reply Dockerfile]: Reverted

File: Dockerfile (hunk 4 above)
Comment (sd4v1d) [reply Dockerfile]: Reverted

File: Dockerfile (hunk 5 above)
Comment (sd4v1d) [reply Dockerfile]: Yes, it was just for debugging. Removed

File: configs/server/config.js (hunk 7 above)
Comment (sd4v1d) [reply configs/server/config.js:162]: Reverted.

File: configs/server/config.js (hunk 8 above)
Comment (sd4v1d) [reply configs/server/config.js:164]: Reverted

File: docker-entrypoint.sh (hunk 9 above)
Comment (sd4v1d) [reply docker-entrypoint.sh:1]: Bash has better conditional tests implementation. Bash is already part of docker image so no additional installation is required.

File: docker-entrypoint.sh (hunk 10 above)
Comment (sd4v1d) [reply docker-entrypoint.sh:17]: Actually, this is it is not in relations with new config mgmt, which, I hope, will resolve this "dirty workaround". 
Since .ts filetype is used, server app also has to be build, and if there is change in config.js file, it has to be rebuild again. This check is added to avoid unnecessary server app rebuilding every time docker image is started. There is comment in the docker-entrypoint.sh file which explains how this works. I really hope that there is better solution and this is only temporary workaround.

**Comment by misi** [comment]
@sd4v1d I run through only. I don't want to block your work..
